### PR TITLE
API doc

### PR DESCRIPTION
**Pull request by IndraGunawan**
i add API doc using https://github.com/FriendsOfPHP/sami

i have a doubt about location of API Doc link at nav menu or inside documentation page.

any advices are welcome

### CODE REVIEW

**Comment by antonmedv** [comment]
Thanks for PR. But i don't see why this is needed? What purposes of this?
Now we have this: https://github.com/deployphp/docs/blob/master/api.md

**Comment by IndraGunawan** [comment]
i think that api just refer how to use deployer in simple steps.

but this is a documentation of all the classes or methods of the project
same like https://getcomposer.org/apidoc/master/index.html

**Comment by antonmedv** [comment]
Ok, let me review this PR.

File: build/api [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+# If no path supplied then use current directory
+if [ -z ${base} ]; then
+    base="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )/.."
+fi
+
+sami=${base}/build/sami
+
+# Download composer if not exists
+if [ ! -f ${sami}/composer.phar ]; then
```
Comment (antonmedv) [inline build/api]: Where is /usr/local/bin/composer

Comment (IndraGunawan) [inline build/api]: that is not using global composer. just download the composer.phar if needed

Comment (antonmedv) [inline build/api]: that will slow down deployment

Comment (IndraGunawan) [inline build/api]: fixed

**Comment by antonmedv** [comment]
Ok, thanks better.  Ill test rest tomorrow, need to work.

**Pull request closed without merging**